optimade/server/middleware.py

```diff
@@ -447,7 +447,8 @@ async def dispatch(self, request: Request, call_next):
             if not isinstance(chunk, bytes):
                 chunk = chunk.encode(charset)
             body += chunk
-        body = body.decode(charset)
+        if response.raw_headers[1][1] == b"application/vnd.api+json":
+            body = body.decode(charset)
```

Comment:
Is this always guaranteed to be at `[1][1]`? Probably better to check via the header keys.

Reply:
Good point, I have changed the code, so it now loops over all entries in the header.